clients/python/tests/test_client.py

```diff
@@ -230,6 +230,39 @@ async def test_update_preserves_model_info(client: ModelRegistry):
     assert updated_ma.model_format_version == model_fmt_version
 
 
+@pytest.mark.e2e
+async def test_update_model_artifact_uri_reuse(client: ModelRegistry):
```

Comment:
Nit: this test might be better named as 
```suggestion
async def test_update_existing_model_artifact(client: ModelRegistry):
```
as __reuse__ isn't too clear about what's being done with it.